wolfcrypt/src/chacha.c

```diff
@@ -265,15 +267,40 @@ static void wc_Chacha_encrypt_bytes(ChaCha* ctx, const byte* m, byte* c,
 
     output = (byte*)temp;
 
+    /* handle left overs */
+    if (ctx->left > 0) {
+        wc_Chacha_wordtobyte(temp, ctx->X); /* recreate the stream */
+        output = (byte*)temp + CHACHA_CHUNK_BYTES - ctx->left;
+        for (i = 0; i < bytes && i < ctx->left; i++) {
+            c[i] = m[i] ^ output[i];
+        }
+        ctx->left = ctx->left - i;
+
+        /* Used up all of the stream that was left, increment the counter */
+        if (ctx->left == 0) {
+            ctx->X[CHACHA_IV_BYTES] = PLUSONE(ctx->X[CHACHA_IV_BYTES]);
```

Comment:
Are you sure this is the right spot?  What about this case but then 64 bytes are still needed.  If you look below the plus one is called again before any bytes are consumed.

Reply:
Please consider the case where the user asks for 1 byte and then 128 bytes.  On the 2nd call, line 281 will do a plus one once the first 64 bytes are consumed with 65 bytes remaining.  Line 291 will be skipped (since remaining is > 64) and the next line will be 303, another plus one without any bytes consumed in-between if I'm reading this right.